src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/LSPEditorFeatureDetector.cs

```diff
@@ -1,13 +1,11 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
-#nullable disable
-
 namespace Microsoft.CodeAnalysis.Razor.Workspaces
 {
     internal abstract class LSPEditorFeatureDetector
     {
-        public abstract bool IsLSPEditorAvailable(string documentMoniker, object hierarchy);
+        public abstract bool IsLSPEditorAvailable(string documentMoniker, object? hierarchy);
```

Comment:
Shouldn't `documentMoniker` here be nullable too? Surprised there isn't a compiler warning for that.. I guess it allows non-null to null, but not the other way.

Reply:
This class is impemented on VsWindows and VsMac side. As you mentioned before, on VsWindows it is clear, that it can be null. However, on VsMac there is no explicit null check and by looking at data flow of this parameter, all methods that accept it later don't expect null either. Making it nullable here force us add a bunch more nullable annotations in many other places. I am not sure, if it even ment to be nullable on mac side then. You can take a look yourself to make a final decision

> it allows non-null to null, but not the other way

Possible values of `string?` contain all values of `string` + `null`, so it is expanding cast. Kinda similar to how you can implicitly cast `int` to `long`, but not the other way around

Reply:
I took a look, and will push up a commit to null annotate this, and add a null check to the VS Mac override. There were no other flow on effects.

My stance on this is that the legacy code is probably correct as-written, in response to some past bug, but poorly annotated. If VS Win did a null check, then I assume at some point there could be a null passed in. The fact that VS Mac didn't have a null check just means VS Mac didn't hit that bug, but usage of VS Mac is smaller than VS Win, so its not very surprising.

Reply:
(obviously I pushed the commit before adding that comment.. sorry.. forgot to hit the submit button 🤦‍♂️)